### PR TITLE
add location validation for tabular reports

### PR DESCRIPTION
**Pull request by calellowitz**
##### SUMMARY

This adds backend validation for location access to tabular report downloads.

### CODE REVIEW

File: custom/icds_reports/views.py [hunk 1]
```diff
@@ -884,6 +884,15 @@ def post(self, request, *args, **kwargs):
 
         location = request.POST.get('location', '')
 
+        if not location and not request.couch_user.has_permission(
+                self.kwargs['domain'], 'access_all_locations'
+        ):
+            return HttpResponseBadRequest()
```
Comment (orangejenny) [inline custom/icds_reports/views.py]: nitpick: wouldn't a 403 make more sense for both of these? that's what `location_restricted_response` does.

Comment (calellowitz) [reply custom/icds_reports/views.py]: Yeah, that sounds right. Fixed

**Review by orangejenny — APPROVED** [review]

File: custom/icds_reports/views.py [hunk 2]
```diff
@@ -2049,6 +2058,11 @@ def post(self, request, *args, **kwargs):
         year = int(request.POST.get('year', None))
         selected_date = date(year, month, 1).strftime('%Y-%m-%d')
 
+        if state_id and not user_can_access_location_id(
```
Comment (orangejenny) [inline custom/icds_reports/views.py]: Why doesn't this also check for full org permissions if state_id isn't present?

Comment (calellowitz) [reply custom/icds_reports/views.py]: These only exist at the state level, you can't download nationally so leaving that out will cause the check at https://github.com/dimagi/commcare-hq/pull/26641/files/b7ac5540601cdf914de7058f64f851957763e2d0#diff-129264561592e648289c087ab1350b9bR2067 to fail

**Review by orangejenny — APPROVED** [review]

**Pull request merged**